File: src/main/java/gg/mineral/server/network/packet/play/clientbound/SetExperiencePacket.java
package gg.mineral.server.network.packet.play.clientbound;

import gg.mineral.server.network.packet.Packet;
import io.netty.buffer.ByteBuf;

public record SetExperiencePacket(float experienceBar, short level, short totalExperience) implements Packet.OUTGOING {
    @Override
    public void serialize(ByteBuf os) {
        os.writeFloat(experienceBar);
        os.writeShort(level);
        os.writeShort(totalExperience);
    }

    @Override
    public byte getId() {
        return 0x1F;
    }
}

File: server/src/main/java/gg/mineral/server/network/packet/play/clientbound/SetExperiencePacket.kt
package gg.mineral.server.network.packet.play.clientbound

import gg.mineral.api.network.packet.Packet
import io.netty.buffer.ByteBuf

@JvmRecord
data class SetExperiencePacket(val experienceBar: Float, val level: Short, val totalExperience: Short) :
    Packet.OUTGOING {
    override fun serialize(os: ByteBuf) {
        os.writeFloat(experienceBar)
        os.writeShort(level.toInt())
        os.writeShort(totalExperience.toInt())
    }

    override fun getId(): Byte {
        return 0x1F
    }
}
